### PR TITLE
Feature/report batch item failures sqs

### LINKED ISSUE #372: [Improvement] Report batch item failures SQS
Lambda has build in wonderful retrying logic, related to SQS, described [here](https://docs.aws.amazon.com/lambda/latest/dg/example_serverless_SQS_Lambda_batch_item_failures_section.html)
To use these convenient code examples, we're required to enable Report batch item failures on sqs lambda trigger.
My suggestion, is to add possibility to enable this on lambda sqs trigger in AWS syndicate.

### PR DESCRIPTION
**Pull request by blbulyandavbulyan**
# Why ?
This pull request closes #372, it gives possibility to use this [code](https://docs.aws.amazon.com/lambda/latest/dg/example_serverless_SQS_Lambda_batch_item_failures_section.html), because without these changes report batch failures is disabled, and the only way to enable it is to manual interaction with AWS console.
The subject property here is this (in lambda triggers in AWS console):
image
It's highlighted with red color. 
# What I changed ?
I modified classes related to maven plugin, and also added additional property with default value false in SqsTriggerEventSource java annotation. Also modified two python files, related to processing properties from sqs triggers resources, with backward compatibility.
# Test plan
* create sqs queue and attach it via SqsEvents and SqsTriggerEventSource annotations
* SqsTriggerEventSource has new property enableBatchReportFailures, which is by default set to false in the annotation declaration, so there should be no compilation errors
* execute `syndicate build && syndicate deploy` commands, the property should be set to "No" in the aws console, as in the screenshot above
* change enableBatchReportFailures to true in your annotation for sqs handler class, and execute `syndicate build && syndicate update` commands, now the property should be set to "Yes"
* execute `syndicate clean && syndicate build && syndicate deploy`, the property still should be set to "Yes" in AWS console

### CODE REVIEW

File: syndicate/core/resources/lambda_resource.py [hunk 1]
```diff
@@ -842,6 +842,7 @@ def _create_sqs_trigger_from_meta(self, lambda_name, lambda_arn, role_name,
                                       trigger_meta):
         validate_params(lambda_name, trigger_meta, SQS_TRIGGER_REQUIRED_PARAMS)
         target_queue = trigger_meta['target_queue']
+        enable_batch_report_failures = trigger_meta.get("enable_batch_report_failures", False)
```
Comment (mkutsybala) [inline syndicate/core/resources/lambda_resource.py:845]: Please stick with lines of code no longer than 80 symbols

File: syndicate/connection/lambda_connection.py [hunk 2]
```diff
@@ -498,14 +501,19 @@ def update_code_source(self, lambda_name, s3_bucket, s3_key,
...
+                            batch_window=None, filters: Optional[List] = None,
+                            enable_batch_report_failures=None):
         params = dict(
             UUID=uuid, FunctionName=function_name, BatchSize=batch_size
         )
         if batch_window is not None:
             params['MaximumBatchingWindowInSeconds'] = batch_window
         if filters is not None:
             params['FilterCriteria'] = {'Filters': filters}
+        if enable_batch_report_failures is not None and enable_batch_report_failures:
```
Comment (mkutsybala) [inline syndicate/connection/lambda_connection.py:513]: Please stick with lines of code no longer than 80 symbols
This line of code will work fine and be more Pythonic just like that
"if enable_batch_report_failures:"

**Comment by mkutsybala** [comment]
Hi!
I've reviewed the solution from a Python/syndicate perspective and received feedback related to Java, in general, it's a working solution, but the solution has to be compliant with AWS convention, hence we should work with a list of strings instead of a boolean(flag), so for lambdas with runtime Python/NodeJS it should be key-value pair "function_response_types": [string] for Java runtime it should be enum FunctionResponseTypes with predefined value 'ReportBatchItemFailures'. In the syndicate, this parameter has to be passed to the boto3 method only if the list is not empty.

File: syndicate/connection/lambda_connection.py (hunk 2 above)
Comment (blbulyandavbulyan) [reply syndicate/connection/lambda_connection.py]: I optimized this if statement, and another one(which is the same, but in different function)

File: syndicate/core/resources/lambda_resource.py (hunk 1 above)
Comment (blbulyandavbulyan) [reply syndicate/core/resources/lambda_resource.py]: I moved method parameters to new line

**Comment by blbulyandavbulyan** [comment]
And also I replaced boolean variable to list of strings in python code, and to enum array in java code

**Comment by blbulyandavbulyan** [comment]
I added some custom serialization  logic to JSON in my java enum, because in java code style allows only SUCH_KIND_OF_NAMES for enums

**Review by bohdan-onsha — APPROVED** [review]

**Pull request merged**